src/pages/Chat/steps/LivingSituationInput.tsx

```diff
@@ -58,15 +68,78 @@ const validationSchema = yup
       .oneOf(Object.keys(ApartmentType))
       .required(),
     size: yup
-      .number()
-      .moreThan(0)
-      .lessThan(250, 'TOO BIG APARTMENT') // TODO improve message
-      .required(),
+      .mixed()
+      .test({
+        name: 'isntTooBig',
+        test: (value) => !isNaN(Number(value)) && value < 250,
+        message: 'TOO BIG APARTMENT',
+      })
+      .test({
+        test: (value) =>
+          value === '' || (!isNaN(Number(value)) && Number(value) > 0),
+        message: 'MUST BE VALID NUMBER',
+      })
+      .test({
+        test: (value) => value !== '',
+        message: 'noop',
+      }),
   })
+
   .required()
 
-const isDone = (values: Partial<LivingSituationState> = {}) =>
-  validationSchema.isValidSync(values)
+const isDone = (values: Partial<LivingSituationState> = {}) => {
+  try {
+    validationSchema.validateSync(values)
+    return true
+  } catch (e) {
+    return false
+  }
+}
+const getValidationError = (
+  values: Partial<LivingSituationState> = {},
+): [string, string] | null => {
+  try {
+    validationSchema.validateSync(values)
+    return null
+  } catch (e) {
+    return [e.params.path, e.message]
+  }
+}
+
+const hasValidationErrorForKey = (key: keyof LivingSituationState) => (
+  validationError: yup.ValidationError,
+) =>
+  allPass([
+    pipe(
+      isNil,
+      not,
+    ),
+    pipe(
+      pathOr('', [0]),
+      equals(key),
+    ),
+    pipe(
+      pathOr('', [1]),
+      equals('noop'),
+      not,
+    ),
+  ])(validationError)
+
+const renderValidationError = (key: keyof LivingSituationState) => (
```

Comment:
Isn't this technically a react component?

Reply:
Not really, it's a curried functions and afaik you cant curry react components

Reply:
But you don't need to curry it do you? on line 251 you have all the data, so you could just make it into a react component? or am I not getting something here?